Documentation/Contributors/CodingGuide/README.md

```diff
@@ -0,0 +1,660 @@
+
+## Formatting
+
+In general, format new code the same as the existing code.
+
+* Use four spaces for indentation.  Do not use tab characters.
+* Do not include trailing whitespace.
+* Put `{` on the same line as the previous statement:
+```javascript
+var defaultValue = function(a, b) {
+   // ...
+};
+
+if (!defined(result)) {
+   // ...
+}
+```
+* Use parenthesis judiciously, e.g.,
+```javascript
+var foo = x > 0.0 && y !== 0.0;
+```
+is better written as
+```javascript
+var foo = (x > 0.0) && (y !== 0.0);
+```
+* Use vertical whitespace to separate functions and to group related statements within a function, e.g.,
+```javascript
+var Model = function(options) {
+    // ...
+    this._allowPicking = defaultValue(options.allowPicking, true);
+
+    this._ready = false;
+    this._readyPromise = when.defer();
+    // ...
+};
+```
+* Use single quotes, `'`, instead of double quotes, `"`.  `"use strict"` is an exception and should use double quotes.
+
+## Units
+
+* Use meters for distances.
+* Use radians for angles, except for explicit `Degrees` functions such as `Cartesian3.fromDegrees`.
+* Use seconds for time durations.
+
+## Basic Code Construction
+
+* To avoid type coercion, test for equality with `===` and `!==`, e.g.,
+```javascript
+var f = 1.0;
+
+if (f === 1.0) {
+    // ...
+}
+
+if (f !== 1.0) {
+    // ...
+}
+```
+* To aid the human reader, append `.0` to whole numbers intended to be floating-point values, e.g., unless `f` is an integer,
+```javascript
+var f = 1;
+```
+is better written as
+```javascript
+var f = 1.0;
+```
+* Declare variables where they are first used.  For example,
+```javascript
+var i;
+var m;
+var models = [ /* ... */ ];
+var length = models.length;
+for (i = 0; i < length; ++i) {
+    m = models[i];
+    // Use m
+}
+```
+is better written as
+```javascript
+var models = [ /* ... */ ];
+var length = models.length;
+for (var i = 0; i < length; ++i) {
+    var m = models[i];
+    // Use m
+}
+```
+* Variables have function-level, not block-level scope.  Do not rely on variable hoisting, i.e., using a variable before it is declared, e.g.,
+```javascript
+console.log(i); // i is undefined here.  Never use a variable before it is declared.
+var i = 0.0;
+```
+* Use `undefined` instead of `null`.
+* Test if a variable is defined using Cesium's `defined` function, e.g.,
+```javascript
+var v = undefined;
+if (defined(v)) {
+    // false
+}
+
+var u = {};
+if (defined(u)) {
+    // true
+}
+```
+* Use Cesium's `definedNotNull` function to test the return of `String.match()` and JSON properties.
+* Use Cesium's `freezeObject` function to create enums, e.g.,
+```javascript
+/*global define*/
+define([
+        '../Core/freezeObject'
+    ], function(
+        freezeObject) {
+    "use strict";
+
+    return freezeObject({
```

Comment:
I would suggest a different example here.  Normally we would declare a file-scoped `var` for documentation reasons, and then `return freezeObject(Blah);` at the end.  (this example is from a private enum)